chainstate/src/detail/transaction_verifier/error.rs

```diff
@@ -86,6 +97,8 @@ pub enum ConnectTransactionError {
     SerializationInvariantError(Id<Block>),
     #[error("Timelock rules violated")]
     TimeLockViolation,
+    #[error("Utxo invariant broken")]
+    UtxoInvariantBroken,
```

Comment:
Instead of removing the string and keeping it bare, please add the parent error type so that it can be see. Strings are not good enough because they can only be found with inspection. Hiding errors is not good because it can never be found unless some good logging was done. See for example how CheckBlockError works.